lib/addressable/uri.rb

```diff
@@ -2356,5 +2363,10 @@ def split_path(path)
       splitted << EMPTY_STR if path.end_with? SLASH
       splitted
     end
+
+    def reset_uri_string_and_hash
```

Comment:
Actually, I think I'd prefer a different method name too. The only thing these two have in common is that they're essentially composite values for the entire URI. Method name should probably reflect that rather than being quite so literal. Especially since it's very possible that there would be future additions to the list and in Addressable 3, subclass overrides to this method.

`remove_composite_instance_variables` seems a little too verbose, maybe `remove_composite_values`?